addon/components/connect.js

```diff
@@ -14,13 +14,38 @@ var connect = function(mapStateToComputed, mapDispatchToActions) {
             });
             return props;
         };
+
         var mapDispatch= function(dispatch) {
             var actions = [];
             Object.keys(finalMapDispatchToActions(dispatch)).forEach(function(key) {
                 actions.push(key);
             });
             return actions;
         };
+
+        /**
+         * Transform a key into its private equivalent by prepending an underscore
+         */
+        var getPrivateKey = function(key) {
+            return `_${key}`;
```

Comment:
Any thoughts on storing the keys on the component itself vs. a private Ember.Object property like `_props` (ala ember-data's `_data` property)? In the latter, no key transformation is necessary which seems like a plus. Plus all the data is stored together (and eliminates issues where component users try to use an underscored version of a property for their own purposes and might conflict).